File: doc/td/ceremony-tresorier/src/services/memberService.js
// src/services/memberService.js
import api from './api';

const API_URL = import.meta.env.VITE_API_URL;

export const memberService = {
  getMembers: () => api.get(`${API_URL}/members`),
  
  getMemberById: (id) => api.get(`${API_URL}/members/${id}`),
  
  createMember: (memberData) => api.post(`${API_URL}/members`, memberData),
  
  updateMember: (id, memberData) => api.put(`${API_URL}/members/${id}`, memberData),
  
  deleteMember: (id) => api.delete(`${API_URL}/members/${id}`),
  
  searchMembers: (query) => api.get(`${API_URL}/members/search?query=${query}`),
  
  getActiveMembers: () => api.get(`${API_URL}/members/active`),
};

File: doc/td/demo/src/main/java/com/ceremonie/demo/services/interfaces/MemberService.java
package com.ceremonie.demo.services.interfaces;


import java.util.List;

import com.ceremonie.demo.dto.request.CreateMemberRequest;
import com.ceremonie.demo.dto.request.UpdateMemberRequest;
import com.ceremonie.demo.dto.response.MemberResponse;

public interface MemberService {
    MemberResponse createMember(CreateMemberRequest request);
    MemberResponse updateMember(Long id, UpdateMemberRequest request);
    MemberResponse getMemberById(Long id);
    MemberResponse getMemberByMemberNumber(String memberNumber);
    List<MemberResponse> getAllMembers();
    List<MemberResponse> getActiveMembers();
    List<MemberResponse> searchMembers(String search);
    void deleteMember(Long id);
    Long countActiveMembers();
    String generateMemberNumber();
}
